guides/micronaut-cloud-secrets-azure/micronaut-cloud-secrets-azure.adoc

```diff
@@ -0,0 +1,194 @@
+common:header.adoc[]
+
+common:requirements.adoc[]
+
+common:azure-account-paid.adoc[]
+
+external:micronaut-cloud-database-base/create-app.adoc[]
+
+== Micronaut Azure Key Vault Dependencies
+
+Add the following dependency:
+
+:dependencies:
+
+dependency:micronaut-azure-secret-manager[groupId=io.micronaut.azure]
+
+:dependencies:
+
+== Update the Application configuration to use Azure Key Vault secrets
+
+=== Add the Azure Key Vault secret placeholders to the Application Configuration
+[,diff]
+.src/main/resources/application.yml
+----
+micronaut:
+  application:
+    name: micronautguide
+
+datasources:
+  default:
++   username: ${JDBC-USER}
++   password: ${JDBC-PASSWORD}
+    dialect: MYSQL
+    driverClassName: ${JDBC_DRIVER:com.mysql.cj.jdbc.Driver}
```

Comment:
underscore is used here, is that correct?

Reply:
Dashes are used, because of the Azure naming convention that does not allow underscores. There is a note below this section in the Guide covering this Azure restriction.

`The Azure Key Vault secrets naming convention does not allow "_" to be used as a word joiner. Therefore, when adding JDBC secret placeholders, you should use "-" instead (e.q. JDBC-USER, JDBC-PASSWORD).`

Reply:
<img width="738" alt="Underscore in Azure secret names" src="https://user-images.githubusercontent.com/4265345/191248299-44a80a0f-2a1d-40c5-b908-9a3dda488a3d.png">

Reply:
I am aware, but the reference above an underscore is used

Reply:
Ah, I see your point.

Reply:
Cover in https://github.com/micronaut-projects/micronaut-guides/pull/1099/commits/6a896fc232dd1a42e605a664cb7488b9ffbbe258.